### DISCUSSION TITLE
Half Duplex connection using 2 different IP on server for download and upload

### DISCUSSION THREAD

**Discussion by persian-michael-scott** [XTLS/Xray-core#3490] [Ideas]
in real world scenario when you open a website for example first few packets are upload and the rest will be download packets.
iran GFW is using traffic analysis to block certain IPs and when we have a lot of traffic up and down on a single ip they can detect and block it.
what if we use 2 separate ip on server one being used only for upload and one will be used only for download . then there is no way to block the ip based in traffic analysis. 
and of course both client and server should support this feature.
i don't know how practical it is to implement this or there is a technical networking issue if we want to implement this.
if this feature is feasible it should work on any type of config including vmess , velss ... .

**Comment by mmmray** [comment]
the problem statement is known, and I think it is agreed that it's a good idea. There are quite a few implementations of this outside of xray.

Some previous discussions:

* https://github.com/XTLS/Xray-core/discussions/3482 -- this needs some feedback
* https://github.com/XTLS/Xray-core/discussions/2320
* there are more, but of course I can't find them now

**Reply by persian-michael-scott** [reply]
thanks for answering.
yes i am actually using this type of splitting with https://github.com/radkesvat/WaterWall half duplex feature . but the problem is you need another domestic server that is not behind gfw . then create a tunnel between main server and this relay server with half duplex.
if it could be available in xray then there is no need for domestic server and every non-expert user can use it without knowing whats happening behind the scene.
